rust/src/array.rs

```diff
@@ -210,6 +210,48 @@ macro_rules! def_primitive_array {
                 n
             }
 
+            pub fn add(&self, other: &PrimitiveArray<$native_ty>) -> PrimitiveArray<$native_ty> {
+                self.math_helper(other, |a, b| a + b)
+            }
+
+            pub fn subtract(
+                &self,
+                other: &PrimitiveArray<$native_ty>,
+            ) -> PrimitiveArray<$native_ty> {
+                self.math_helper(other, |a, b| a - b)
+            }
+
+            pub fn multiply(
```

Comment:
does this work for boolean types as well?

Reply:
No, this is only implemented for the numeric types (as with the existing `min` and `max` methods).